pkg/operator/client/deploy.go

```diff
@@ -144,75 +192,56 @@ func (c *Client) ensureResourcesPresent(deployArgs operatortypes.DeployAppArgs)
+
+			if len(applyStdout) > 0 {
+				deployRes.applyResult.multiStdout = append(deployRes.applyResult.multiStdout, applyStdout)
 			}
-			if len(dryrunStderr) > 0 {
-				deployRes.dryRunResult.multiStderr = append(deployRes.dryRunResult.multiStderr, dryrunStderr)
+			if len(applyStderr) > 0 {
+				deployRes.applyResult.multiStderr = append(deployRes.applyResult.multiStderr, applyStderr)
 			}
 
-			if dryRunErr != nil {
-				logger.Infof("stdout (dryrun) = %s", dryrunStdout)
-				logger.Infof("stderr (dryrun) = %s", dryrunStderr)
-				logger.Infof("error: %s", dryRunErr.Error())
+			if applyErr != nil {
+				logger.Infof("stdout (apply) = %s", applyStdout)
+				logger.Infof("stderr (apply) = %s", applyStderr)
+				logger.Infof("error: %s", applyErr.Error())
 
-				deployRes.dryRunResult.hasErr = true
+				deployRes.applyResult.hasErr = true
 				return &deployRes, nil
 			}
 
 			if resource.DecodeErrMsg == "" {
-				logger.Infof("dry run applied resource %s/%s/%s/%s in namespace %s", group, version, kind, name, namespace)
+				logger.Infof("applied resource %s/%s/%s/%s in namespace %s", group, version, kind, name, namespace)
 			} else {
-				logger.Info("dry run applied unidentified resource.")
+				logger.Info("applied unidentified resource.")
 			}
-		}
-	}
 
-	for _, resource := range resources {
-		group := resource.GetGroup()
-		version := resource.GetVersion()
-		kind := resource.GetKind()
-		name := resource.GetName()
-
-		namespace := resource.GetNamespace()
-		if namespace == "" {
-			namespace = c.TargetNamespace
-		}
-
-		if resource.DecodeErrMsg == "" {
-			logger.Infof("applying resource %s/%s/%s/%s in namespace %s", group, version, kind, name, namespace)
-		} else {
-			logger.Infof("applying unidentified resource. unable to parse error: %s", resource.DecodeErrMsg)
-		}
-
-		applyStdout, applyStderr, applyErr := kubernetesApplier.ApplyCreateOrPatch(namespace, deployArgs.AppSlug, []byte(resource.Manifest), false, deployArgs.Wait, deployArgs.AnnotateSlug)
-
-		if len(applyStdout) > 0 {
-			deployRes.applyResult.multiStdout = append(deployRes.applyResult.multiStdout, applyStdout)
-		}
-		if len(applyStderr) > 0 {
-			deployRes.applyResult.multiStderr = append(deployRes.applyResult.multiStderr, applyStderr)
-		}
-
-		if applyErr != nil {
-			logger.Infof("stdout (apply) = %s", applyStdout)
-			logger.Infof("stderr (apply) = %s", applyStderr)
-			logger.Infof("error: %s", applyErr.Error())
-
-			deployRes.applyResult.hasErr = true
-			return &deployRes, nil
-		}
+			if resource.ShouldWaitForReady() {
+				logger.Infof("waiting for resource %s/%s/%s/%s in namespace %s to be ready", group, version, kind, name, namespace)
+				err := appstate.WaitForResourceToBeReady(namespace, name, resource.GVK)
+				if err != nil {
+					logger.Infof("failed to wait for resource %s/%s/%s/%s in namespace %s to be ready", group, version, kind, name, namespace)
+					return nil, errors.Wrap(err, "failed to wait for resource to be ready")
```

Comment:
why not use the same error message from the log above?

Reply:
makes sense since it will show in the UI for debugging